app/scripts/move-popup/dimMoveItemProperties.directive.js

```diff
@@ -18,70 +18,39 @@
-        '       <span ng-if="stat.bar && stat.value < stat.equippedStatsValue && stat.comparable" class="stat-box-inner lower-stats" dim-percent-width="(stat.equippedStatsValue - stat.value) / stat.maximumValue"></span>',
-        '       <span ng-if="stat.bar && stat.value > stat.equippedStatsValue && stat.comparable" class="stat-box-inner" dim-percent-width="stat.equippedStatsValue / stat.maximumValue"></span>',
-        '       <span ng-if="stat.bar && stat.value > stat.equippedStatsValue && stat.comparable" class="stat-box-inner higher-stats" dim-percent-width="(stat.value - stat.equippedStatsValue) / stat.maximumValue"></span>',
-        '       <span ng-if="!stat.bar && (!stat.equippedStatsName || stat.comparable)" ng-class="{ \'higher-stats\': (stat.value > stat.equippedStatsValue), \'lower-stats\': (stat.value < stat.equippedStatsValue)}">{{ stat.value }}</span>',
-        '     </span>',
-        '     <span class="stat-box-val stat-box-cell" ng-class="{ \'higher-stats\': (stat.value > stat.equippedStatsValue && stat.comparable), \'lower-stats\': (stat.value < stat.equippedStatsValue && stat.comparable)}" ng-show="{{ stat.bar }}">{{ stat.value }}',
-        '       <span ng-if="stat.bar && vm.itemQuality && stat.qualityPercentage.min" ng-style="stat.qualityPercentage.min | qualityColor:\'color\'">({{ stat.qualityPercentage.range }})</span>',
-        '     </span>',
-        '  </div>',
-        '  <div class="stat-box-row" ng-if="vm.item.quality && vm.item.quality.min">',
-        '    <span class="stat-box-text stat-box-cell">Stats quality</span>',
-        '    <span class="stat-box-cell" ng-style="vm.item.quality.min | qualityColor:\'color\'">{{ vm.item.quality.range }} of max roll</span>',
-        '  </div>',
-        '</div>',
-        '<div class="item-details item-perks" ng-if="vm.item.talentGrid && vm.itemDetails">',
-        '  <dim-talent-grid dim-talent-grid="vm.item.talentGrid" dim-infuse="vm.infuse(vm.item, $event)"/>',
-        '</div>',
-        '<div class="item-details item-objectives" ng-if="vm.item.objectives.length && vm.itemDetails">',
-        '  <div class="objective-row" ng-repeat="objective in vm.item.objectives track by $index" ng-class="{\'objective-complete\': objective.complete, \'objective-boolean\': objective.boolean }">',
-        '     <div class="objective-checkbox"><div></div></div>',
-        '     <div class="objective-progress">',
-        '       <div class="objective-progress-bar" dim-percent-width="objective.progress / objective.completionValue"></div>',
-        '       <div class="objective-description">{{ objective.description || (objective.complete ? \'Complete\' : \'Incomplete\') }}</div>',
-        '       <div class="objective-text">{{ objective.progress }} / {{ objective.completionValue }}</div>',
-        '     </div>',
-        '  </div>',
-        '</div>'
-      ].join('')
+      templateUrl: 'templates/moveItemProperties.html'
```

Comment:
That stinks. Do you have a reason for keeping the template inline? I'm not sure about performance hits, but inline templates this big are a pain to work with 😭